notebooks/transport/http_test.go

```diff
@@ -0,0 +1,210 @@
+package transport
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/influxdata/influxdb/v2/kit/feature"
+	"github.com/influxdata/influxdb/v2/kit/platform"
+	"github.com/influxdata/influxdb/v2/notebooks/service"
+	"github.com/influxdata/influxdb/v2/notebooks/service/mocks"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+)
+
+var (
+	orgStr       = "1234123412341234"
+	orgID, _     = platform.IDFromString(orgStr)
+	idStr        = "4321432143214321"
+	id, _        = platform.IDFromString(idStr)
+	testNotebook = &service.Notebook{
+		OrgID: *orgID,
+		ID:    *id,
+		Name:  "test notebook",
+		Spec: service.NotebookSpec{
+			"hello": "goodbye",
+		},
+	}
+	testReqBody = &service.NotebookReqBody{
+		OrgID: *orgID,
+		Name:  "Test notebook",
+		Spec: service.NotebookSpec{
+			"hello": "goodbye",
+		},
+	}
+)
+
+func TestNotebookHandler(t *testing.T) {
+	t.Parallel()
+
+	var (
+		ctrlr = gomock.NewController(t)
```

Comment:
Is gomock smart enough to handle sharing a controller between multiple test cases?

Reply:
Turns out it definitely isn't! https://pkg.go.dev/github.com/golang/mock/gomock?utm_source=godoc#Controller

I'm surprised sharing the same `t` & controller for all the tests worked as well as it did actually. It seemed to handle all the failures as expected. I updated the code to create a new controller for each `t` now, which - if nothing else - makes the output more specific as to which test fails if any do.